app/controllers/prison/dashboards_controller.rb

```diff
@@ -35,23 +35,36 @@ def search
     cancellations
     processed_visits
 
-    @estate = user_estate
+    @estate = current_estate
+  end
+
+  def switch_estate
+    estate = Estate.find(params[:estate_id])
```

Comment:
Again, this could result in a 404 which would be confusing to the user and when debugging if the estate_id is not valid.